### PR TITLE
Feature/update to polkadot v0 9 23

### PR DESCRIPTION
**Pull request by MRamanenkau**

### CODE REVIEW

File: runtime/cere-dev/Cargo.toml [hunk 1]
```diff
@@ -101,7 +101,7 @@ pallet-erc20 = { version = "2.0.0", default-features = false, path = "../../pall
 pallet-ddc-metrics-offchain-worker = { version = "2.0.0", default-features = false, path = "../../pallets/ddc-metrics-offchain-worker" }
 
 [build-dependencies]
-substrate-wasm-builder = { version = "5.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.22" }
+substrate-wasm-builder = { version = "5.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.23" }
```
Comment (AndreiNavoichyk) [inline runtime/cere-dev/Cargo.toml:104]: Update [try-runtime] section with `pallet-bags-list`

File: runtime/cere-dev/src/lib.rs [hunk 2]
```diff
@@ -958,6 +979,8 @@ impl pallet_contracts::Config for Runtime {
 	type Schedule = Schedule;
 	type AddressGenerator = pallet_contracts::DefaultAddressGenerator;
 	type ContractAccessWeight = pallet_contracts::DefaultContractAccessWeight<RuntimeBlockWeights>;
+	type MaxCodeLen = ConstU32<{ 128 * 1024 }>;
```
Comment (AndreiNavoichyk) [inline runtime/cere-dev/src/lib.rs:982]: Why `128`?

File: runtime/cere/Cargo.toml [hunk 3]
```diff
@@ -100,7 +100,7 @@ pallet-erc20 = { version = "2.0.0", default-features = false, path = "../../pall
 pallet-ddc-metrics-offchain-worker = { version = "2.0.0", default-features = false, path = "../../pallets/ddc-metrics-offchain-worker" }
 
 [build-dependencies]
-substrate-wasm-builder = { version = "5.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.22" }
+substrate-wasm-builder = { version = "5.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.23" }
```
[the remark this answers is not in the document]
Comment (MRamanenkau) [reply runtime/cere/Cargo.toml:103]: We already had `pallet-bags-list/try-runtime` in `try-runtime`

File: runtime/cere/src/lib.rs [hunk 4]
```diff
@@ -635,6 +635,28 @@ impl onchain::BoundedConfig for OnChainSeqPhragmen {
 	type TargetsBound = ConstU32<2_000>;
 }
 
+impl pallet_election_provider_multi_phase::MinerConfig for Runtime {
```
[the remark this answers is not in the document]
Comment (MRamanenkau) [reply runtime/cere/src/lib.rs:638]: Added

File: runtime/cere/src/lib.rs [hunk 5]
```diff
@@ -957,6 +978,8 @@ impl pallet_contracts::Config for Runtime {
 	type Schedule = Schedule;
 	type AddressGenerator = pallet_contracts::DefaultAddressGenerator;
 	type ContractAccessWeight = pallet_contracts::DefaultContractAccessWeight<RuntimeBlockWeights>;
+	type MaxCodeLen = ConstU32<{ 128 * 1024 }>;
+	type RelaxedMaxCodeLen = ConstU32<{ 256 * 1024 }>;
```
[the remark this answers is not in the document]
Comment (MRamanenkau) [reply runtime/cere/src/lib.rs:982]: This is the default Substrate value. I'll run the tests to make sure that it fits our smart contracts

File: runtime/cere/src/lib.rs [hunk 6]
```diff
@@ -957,6 +978,8 @@ impl pallet_contracts::Config for Runtime {
 	type Schedule = Schedule;
 	type AddressGenerator = pallet_contracts::DefaultAddressGenerator;
 	type ContractAccessWeight = pallet_contracts::DefaultContractAccessWeight<RuntimeBlockWeights>;
+	type MaxCodeLen = ConstU32<{ 128 * 1024 }>;
```
[the remark this answers is not in the document]
Comment (MRamanenkau) [reply runtime/cere/src/lib.rs:981]: This is the default Substrate value. I'll run the tests to make sure that it fits our smart contracts

File: runtime/cere/src/impls.rs [hunk 7]
```diff
@@ -217,7 +217,9 @@ mod multiplier_tests {
 				fm = next;
 				iterations += 1;
 				let fee =
-					<Runtime as pallet_transaction_payment::Config>::WeightToFee::calc(&tx_weight);
+					<Runtime as pallet_transaction_payment::Config>::WeightToFee::weight_to_fee(
```
[the remark this answers is not in the document]
Comment (MRamanenkau) [reply runtime/cere/src/impls.rs:220]: Added

File: runtime/cere-dev/src/lib.rs [hunk 8]
```diff
@@ -958,6 +979,8 @@ impl pallet_contracts::Config for Runtime {
...
+	type RelaxedMaxCodeLen = ConstU32<{ 256 * 1024 }>;
```
[the remark this answers is not in the document]
Comment (MRamanenkau) [reply runtime/cere-dev/src/lib.rs:983]: This is the default Substrate value. I'll run the tests to make sure that it fits our smart contracts

File: runtime/cere-dev/Cargo.toml (hunk 1 above)
Comment (MRamanenkau) [reply runtime/cere-dev/Cargo.toml:104]: We already had pallet-bags-list/try-runtime in try-runtime

Comment (AndreiNavoichyk) [reply runtime/cere-dev/Cargo.toml:104]: Please, send to me a link to the line. I can not find it.

Comment (MRamanenkau) [reply runtime/cere-dev/Cargo.toml:104]: Sorry. I was checking the wrong place. Added.

File: runtime/cere-dev/src/lib.rs [hunk 9]
```diff
@@ -126,7 +126,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	// and set impl_version to 0. If only runtime
 	// implementation changes and behavior does not, then leave spec_version as
 	// is and increment impl_version.
-	spec_version: 30701,
+	spec_version: 30801,
```
Comment (AndreiNavoichyk) [inline runtime/cere-dev/src/lib.rs:129]: It should be `30800`

File: runtime/cere-dev/src/lib.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (AndreiNavoichyk) [reply runtime/cere-dev/src/lib.rs:983]: Do you have updates here?

File: runtime/cere-dev/src/lib.rs (hunk 9 above)
Comment (MRamanenkau) [reply runtime/cere-dev/src/lib.rs]: Fixed

File: runtime/cere-dev/src/lib.rs (hunk 2 above)
Comment (MRamanenkau) [reply runtime/cere-dev/src/lib.rs:982]: Could you please check it during the test run?

File: runtime/cere-dev/src/lib.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (MRamanenkau) [reply runtime/cere-dev/src/lib.rs:983]: Could you please check it during the test run?

File: runtime/cere-dev/src/lib.rs (hunk 2 above)
Comment (AndreiNavoichyk) [reply runtime/cere-dev/src/lib.rs:982]: Tested. Works as expected.

File: runtime/cere-dev/src/lib.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (AndreiNavoichyk) [reply runtime/cere-dev/src/lib.rs:983]: Tested. Works as expected.

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**